src/postamble.js

```diff
@@ -509,8 +509,9 @@ if (Module['noInitialRun']) shouldRunNow = false;
 
 #endif // HAS_MAIN
 
-#if EXIT_RUNTIME == 0
-#if USE_PTHREADS
+#if EXIT_RUNTIME
+noExitRuntime = false;
```

Comment:
How about setting it on the var? This will do it:
```diff
diff --git a/src/preamble.js b/src/preamble.js
index 042bb0e9c..41f65c93a 100644
--- a/src/preamble.js
+++ b/src/preamble.js
@@ -20,7 +20,7 @@ out = err = function(){};
 #endif
 
 {{{ makeModuleReceiveWithVar('wasmBinary') }}}
-{{{ makeModuleReceiveWithVar('noExitRuntime') }}}
+{{{ makeModuleReceiveWithVar('noExitRuntime', undefined, 'false') }}}
 
 #if WASM != 2 && MAYBE_WASM2JS
 #if !WASM2JS
```
Then we emit `var noExitRuntime = Module["noExitRuntime"] || false;`